common/persistence/sql/sqlplugin/history.go

```diff
@@ -25,13 +25,13 @@ package sqlplugin
 import "database/sql"
 
 type (
-	// historyShard is the SQL persistence interface for history shards
-	historyShard interface {
+	// HistoryShard is the SQL persistence interface for history shards
+	HistoryShard interface {
 		InsertIntoShards(rows *ShardsRow) (sql.Result, error)
 		UpdateShards(row *ShardsRow) (sql.Result, error)
 		SelectFromShards(filter *ShardsFilter) (*ShardsRow, error)
-		ReadLockShards(filter *ShardsFilter) (int, error)
-		WriteLockShards(filter *ShardsFilter) (int, error)
+		ReadLockShards(filter *ShardsFilter) (int64, error)
```

Comment:
I think @sergeybykov converted (wanted to convert) all `shardID` to `int32` from `int64`. Please check with him.

Reply:
not really, lots of shard IDs are still either `int` or `int64`
this is a short term solution. in my next PRs, all will be converted to `int32`

Reply:
I only touched those usages where they were being explicitly converted to strings. There are still `int`s and `int64`s left to clean up.

Reply:
@sergeybykov i will carry out the change.